orderbook/usecase/orderbook_usecase_test.go

```diff
@@ -0,0 +1,314 @@
+							},
+						},
+						UnrealizedCancels: orderbookdomain.UnrealizedCancels{
+							AskUnrealizedCancels: osmomath.NewInt(100),
+						},
+					}, true
+				}
+				tokensUsecase.GetSpotPriceScalingFactorByDenomFunc = func(baseDenom, quoteDenom string) (osmomath.Dec, error) {
+					return osmomath.NewDec(10), nil
+				}
+			},
+		},
+		{
+			name: "successful order processing",
+			order: orderbookdomain.Order{
+				TickId:         1,
+				OrderId:        1,
+				OrderDirection: "bid",
+				Owner:          "owner1",
+				Quantity:       "1000",
+				PlacedQuantity: "1500",
+				Etas:           "500",
+				ClaimBounty:    "10",
+				PlacedAt:       "1634764800000",
+			},
+			quoteAsset: orderbookdomain.Asset{Symbol: "ATOM", Decimals: 6},
+			baseAsset:  orderbookdomain.Asset{Symbol: "OSMO", Decimals: 6},
+			setupMocks: func(orderbookRepo *mocks.OrderbookRepositoryMock, tokensUsecase *mocks.TokensUsecaseMock) {
+				orderbookRepo.GetTickByIDFunc = func(poolID uint64, tickID int64) (orderbookdomain.OrderbookTick, bool) {
+					return orderbookdomain.OrderbookTick{
+						TickState: orderbookdomain.TickState{
+							BidValues: orderbookdomain.TickValues{
+								EffectiveTotalAmountSwapped: "500",
+							},
+						},
+						UnrealizedCancels: orderbookdomain.UnrealizedCancels{
+							BidUnrealizedCancels: osmomath.NewInt(100),
+						},
+					}, true
+				}
+				tokensUsecase.GetSpotPriceScalingFactorByDenomFunc = func(baseDenom, quoteDenom string) (osmomath.Dec, error) {
+					return osmomath.NewDec(1), nil
+				}
+			},
+			expectedError: "",
+			expectedOrder: orderbookdomain.LimitOrder{
+				TickId:           1,
+				OrderId:          1,
+				OrderDirection:   "bid",
+				Owner:            "owner1",
+				Quantity:         1000,
+				Etas:             "500",
+				ClaimBounty:      "10",
+				PlacedQuantity:   1500,
+				PlacedAt:         1634,
+				Price:            "1.000001000000000000",
+				PercentClaimed:   "0.333333333333333333",
+				TotalFilled:      600,
+				PercentFilled:    "0.400000000000000000",
+				OrderbookAddress: "someOrderbookAddress",
+				Status:           "partiallyFilled",
+				Output:           "1499.998500001499998500",
+				QuoteAsset:       orderbookdomain.Asset{Symbol: "ATOM", Decimals: 6},
+				BaseAsset:        orderbookdomain.Asset{Symbol: "OSMO", Decimals: 6},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			// Create instances of the mocks
+			orderbookRepo := mocks.OrderbookRepositoryMock{}
+			tokensUsecase := mocks.TokensUsecaseMock{}
+
+			// Setup the mocks according to the test case
+			tc.setupMocks(&orderbookRepo, &tokensUsecase)
+
+			// Initialize the use case with the mocks
+			usecase := orderbookusecase.New(
+				&orderbookRepo,
+				nil,
+				nil,
+				&tokensUsecase,
+				nil,
+			)
+
+			// Call the method under test
+			result, err := usecase.CreateFormattedLimitOrder(1, tc.order, tc.quoteAsset, tc.baseAsset, "someOrderbookAddress")
```

Comment:
PoolID is passed because it is requires as argument, actual function that uses this argument is mocked, thus it can be any number value. To avoid use of constant I have moved it to `tc.poolID` that defauts to `0` at this moment for all test cases.